src/Transmogrification.cpp

```diff
@@ -553,15 +547,130 @@ bool Transmogrification::SuitableForTransmogrification(Player* player, ItemTempl
     if (!IgnoreReqLevel && player->getLevel() < proto->RequiredLevel)
         return false;
 
+    return true;
+}
+
+bool Transmogrification::SuitableForTransmogrification(ObjectGuid::LowType playerGuid, ItemTemplate const* proto) const
+{
+    if (!playerGuid || !proto)
+        return false;
+
+    if (proto->Class != ITEM_CLASS_ARMOR &&
+        proto->Class != ITEM_CLASS_WEAPON)
+        return false;
+
+    // Skip all checks for allowed items
+    if (IsAllowed(proto->ItemId))
+        return true;
+
+    if (!CheckPureProtoRequirements(proto))
+        return false;
+
+    uint32 playerRaceMask;
+    uint32 playerClassMask;
+    TeamId playerTeamId;
+    uint8 playerLevel;
+    std::unordered_map<uint32, uint32> playerSkillValues;
+
+    if (QueryResult resultPlayer = CharacterDatabase.Query("SELECT `race`, `class`, `level` FROM `characters` WHERE `guid` = {}", playerGuid))
```

Comment:
All of this information is cached and doesnt require a query
Check the CharacterCache implementation

Reply:
ah damn, thanks

Reply:
Replaced in new commit bfc4d32c0494e1e78361bb0813be3755ecbe163a